starfish/core/image/_filter/reduce.py

```diff
@@ -0,0 +1,144 @@
+from typing import (
+    Callable,
+    Iterable,
+    MutableMapping,
+    Sequence,
+    Union
+)
+
+import numpy as np
+
+from starfish.core.imagestack.imagestack import ImageStack
+from starfish.core.types import Axes, Clip, Coordinates, Number
+from starfish.core.util import click
+from starfish.core.util.dtype import preserve_float_range
+from ._base import FilterAlgorithmBase
+
+
+class Reduce(FilterAlgorithmBase):
+    """
+    Reduces the dimensions of the ImageStack by applying a function
+    along one or more axes.
+
+    Parameters
+    ----------
+    dims : Axes
+        one or more Axes to project over
+    func : Union[str, Callable]
+        function to apply across the dimension(s) specified by dims.
+        If a function is provided, it should follow the form specified by
+        DataArray.reduce():
+        http://xarray.pydata.org/en/stable/generated/xarray.DataArray.reduce.html
+        If a string is provided, it should correspond to a numpy function that
+        matches the form specified above
+        (i.e., function is resolved: func = getattr(np, func)).
+        Some common examples below:
+        max: maximum intensity projection (applies numpy.amax)
```

Comment:
@kevinyamauchi `max` -> `np.amax` is not consistent with the stated functionality of the documentation using `getattr`, and in the code, it's special cased. Can you resolve the logical discontinuity in the documentation?

Reply:
Ah good point. Fixed.